pkg/client/client.go

```diff
@@ -593,3 +594,19 @@ func (client *DataSetClient) onBufferDrop(buf *buffer.Buffer, status uint32, err
 		)...,
 	)
 }
+
+func (client *DataSetClient) setLastErrorTimestamp(timestamp time.Time) {
+	client.lastErrorMu.Lock()
+	defer client.lastErrorMu.Unlock()
+	client.lastErrorTs = timestamp
+}
+
+func (client *DataSetClient) lastErrorTimestamp() time.Time {
+	client.retryAfterMu.RLock()
+	defer client.retryAfterMu.RUnlock()
+	return client.lastErrorTs
+}
+
+func (client *DataSetClient) isLastRetryableErrorTimedOut() bool {
+	return client.lastErrorTimestamp().Add(client.Config.BufferSettings.RetryMaxElapsedTime).Before(time.Now())
```

Comment:
Should we also clean up / reset this timestamp when timeout is reached (e.g. set it to 0 or nil)? And then also account for that here?

Reply:
I assume we already / will also expose ``RetryMaxElapsedTime`` config option in the OTEL log exporter config, right?

Reply:
Not sure if I understand the point. Why should we reset timestamp?